### PR TITLE
Support create/edit user role on user page

### PR DESCRIPTION
**Pull request by shanghaikid**
image
image
image

### CODE REVIEW

File: client/src/pages/user/CreateUser.tsx [hunk 1]
```diff
@@ -98,6 +120,37 @@ const CreateUser: FC<CreateUserProps> = ({ handleCreate, handleClose }) => {
             key={v.label}
           />
         ))}
+
+        <Typography variant="h5" component="span">
+          {userTrans('roles')}
+        </Typography>
+
+        <FormGroup row>
+          {roles.map((r: any, index: number) => (
```
Comment (nameczz) [inline client/src/pages/user/CreateUser.tsx:129]: Cloud we use string instead of any?

File: client/src/pages/user/Types.ts [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
+  onUpdate: (data: UpdateUserRoleParams) => void;
+  handleClose: () => void;
+  username: string;
+  roles: string[];
 }
 
 export interface CreateUserProps {
   handleCreate: (data: CreateUserParams) => void;
   handleClose: () => void;
+  roles: Option[];
```
Comment (nameczz) [inline client/src/pages/user/Types.ts:29]: Should we rename roles to roleOptions?

File: client/src/pages/user/User.tsx [hunk 3]
```diff
@@ -89,6 +123,35 @@ const Users = () => {
...
+          open: true,
+          type: 'custom',
+          params: {
+            component: (
+              <UpdateUserRole
+                username={selectedUser[0]!.name}
+                onUpdate={onUpdate}
+                handleClose={handleCloseDialog}
+                roles={
+                  users.filter(u => u.name === selectedUser[0].name)[0].roles
```
Comment (nameczz) [inline client/src/pages/user/User.tsx:140]: do we need to check undefined here?

**Review by nameczz — CHANGES_REQUESTED** [review]

File: client/src/pages/user/User.tsx (hunk 3 above)
Comment (shanghaikid) [reply client/src/pages/user/User.tsx:140]: no

**Review by nameczz — APPROVED** [review]

**Pull request merged**